### ISSUE TITLE
Wobbly cards during animations

### ISSUE THREAD

**Issue by BrainCrumbz** [raflop/ToastNotifications#9]
Hi there

while trying out the library in our team, we met a visual issue shown here below:

![basic-example-wobbly](https://cloud.githubusercontent.com/assets/3185573/16582578/036cf39e-42b0-11e6-942e-7278106affa4.gif)

That is just the *BasicUsageExample* running from VS2013 with following changes:

```xaml
<!-- MainWindow.xaml -->
<Window ... xmlns:toast="clr-namespace:ToastNotifications;assembly=ToastNotifications">
...
        <toast:NotificationTray NotificationsSource="{Binding NotificationSource}"
                                PopupFlowDirection="RightDown"  VerticalAlignment="Bottom" HorizontalAlignment="Right" />
```

Especially when fading out, the stack somehow *wobbles*, apparently because the bottom card changes height, and whole stack jumps up and down a little bit. 

During execution, upon generating the first notification, the following message shows up in Immediate window console, not sure if that can be useful:

> System.Windows.Data Error: 2 : Cannot find governing FrameworkElement or FrameworkContentElement for target element. BindingExpression:Path=Icon; DataItem=null; target element is 'VisualBrush' (HashCode=47097466); target property is 'Visual' (type 'Visual')

**Edit:** after some investigation: the issue disappears if you disable all animations and transform (maybe obvious). Also, it is present when `PopupFlowDirection="RightDown"  VerticalAlignment="Bottom"`, while not present when `PopupFlowDirection="RightUp"  VerticalAlignment="Top" `. I guess [this switch](https://github.com/raflop/ToastNotifications/blob/51230b4077ac975f81945a326fc66b145af3d8ff/ToastNotifications/NotificationPopupWindow.xaml.cs#L134) might be related to the issue

For some reason, in our actual application the effect is even more visible:

![application-toast-wobbly](https://cloud.githubusercontent.com/assets/3185573/16583019/09696590-42b3-11e6-9406-7c366b9c8d47.gif)

And there two more observations there:

1. When showing a notification starting from an empty stack (whether it's the absolute first one, or it's the first after all previous faded out), that first notification sits a little bit offscreen to the right. That is until a second notification is added: at that point, the first one proceeds moving to the left and then scrolls up. **Edit:** if it might help, this effects disappears when you completely disable `ScaleX` transform & animations

1. If text is larger than card size, the text is clipped. How to make card larger, or adapt its size to content?

Thanks for your time

**Comment by BrainCrumbz** [comment]
Hi there, anyone available on this? Thank you

**Comment by raflop** [comment]
Hi thanks for your commnets, I will try to fix this, but it requires some major changes in notification panel implementation, because actually it's based on ItemsControl to show notifications, and it has some limitations.

**Comment by BrainCrumbz** [comment]
If you could shed some light on library internals, maybe our team can lend a hand. For this issue, how is the UI composed? What are its parts? How does the layouting work?

**Comment by raflop** [comment]
Hi I fixed problems described here in branch develop. (except that "For some reason, in our actual application the effect is even more visible:" i will fix this soon ) 

Could you check if it works correctly for you?

**Comment by raflop** [comment]
I fixed issue with text.

**Comment by BrainCrumbz** [comment]
Sure!

**Comment by BrainCrumbz** [comment]
Updating our desktop app to 1.3.0:

Still "wobbly" when `PopupFlowDirection="RightDown" VerticalAlignment="Bottom" HorizontalAlignment="Right"`:

![from-bottom-upwards](https://cloud.githubusercontent.com/assets/3185573/16787569/7b70f5f0-489d-11e6-9a0a-a3eaa7f5c0e2.gif)

Still ok when `PopupFlowDirection="RightUp" VerticalAlignment="Top" HorizontalAlignment="Right"`:

![from-top-downwards](https://cloud.githubusercontent.com/assets/3185573/16787618/d950832a-489d-11e6-8853-3a5f8e65dd7b.gif)

Horizontal displacement on first item is there.

Updating basic usage sample to latest from repo, again referencing 1.3.0, yields similar result:

Still "wobbly" when `PopupFlowDirection="RightDown" VerticalAlignment="Bottom" HorizontalAlignment="Right"`:

![basic-from-bottom-upwards](https://cloud.githubusercontent.com/assets/3185573/16787770/440aed3a-489f-11e6-8636-43910dcb5279.gif)

**Comment by raflop** [comment]
Hi,  

Sorry I need to ask :) Did you checkout develop branch?  Because master and nuget don't have my yesterday's fixes. 





W dniu 13 lip 2016, 02:13, o 02:13, użytkownik BrainCrumbz <notifications@github.com> napisał:

**Comment by BrainCrumbz** [comment]
... Good you asked :)

**Comment by BrainCrumbz** [comment]
Yay! 

Basic sample app:

![basic-example-bottom-upwards](https://cloud.githubusercontent.com/assets/3185573/16795800/85a81ffc-48df-11e6-8c46-1bbedb4342d9.gif)

Our custom app:

![custom-app-bottom-upwards](https://cloud.githubusercontent.com/assets/3185573/16795802/8a6cbd0e-48df-11e6-8e02-26da96fe068e.gif)

Larger text now is shown as well. 

Horizontal animation issue now has gone (I guess one of the visible effects now is more horizontal space between cards and window border)

**Comment by raflop** [comment]
I fixed problem with margin. 
All fixes avalaible in nuget 1.4.0

**Comment by BrainCrumbz** [comment]
It looks great! Thanks
